spanner/api/Spanner.Samples.Tests/SpannerFixture.cs

```diff
@@ -0,0 +1,250 @@
+// Copyright 2020 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using Google.Cloud.Spanner.Admin.Database.V1;
+using Google.Cloud.Spanner.Admin.Instance.V1;
+using Google.Cloud.Spanner.Common.V1;
+using Google.Cloud.Spanner.Data;
+using Grpc.Core;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Xunit;
+
+[CollectionDefinition(nameof(SpannerFixture))]
+public class SpannerFixture : IDisposable, ICollectionFixture<SpannerFixture>
+{
+    public string ProjectId { get; set; } = Environment.GetEnvironmentVariable("GOOGLE_PROJECT_ID");
+    // Allow environment variables to override the default instance and database names.
+    public string InstanceId { get; set; } = Environment.GetEnvironmentVariable("TEST_SPANNER_INSTANCE") ?? "my-instance";
+    public string DatabaseId { get; set; } = Environment.GetEnvironmentVariable("TEST_SPANNER_DATABASE") ?? $"my-db-{DateTimeOffset.UtcNow.ToUnixTimeMilliseconds()}";
+    public string BackupDatabaseId { get; set; } = "my-test-database";
+    public string BackupId { get; set; } = "my-test-database-backup";
+    public string ToBeCancelledBackupId { get; set; } = $"my-backup-{DateTimeOffset.UtcNow.ToUnixTimeMilliseconds()}";
+    public string RestoredDatabaseId { get; set; } = $"my-restore-db-{DateTimeOffset.UtcNow.ToUnixTimeMilliseconds()}";
+    public List<string> DatabasesToDelete { get; set; } = new List<string>();
+
+    public SpannerFixture()
+    {
+        // Don't need to cleanup stale Backups and Databases when instance is new.
+        var isExistingInstance = InitializeInstance();
+        if (isExistingInstance)
+        {
+            DeleteStaleBackupsAndDatabases();
+        }
+        InitializeDatabase();
+        InitializeBackup();
+    }
+
+    private bool InitializeInstance()
+    {
+        InstanceAdminClient instanceAdminClient = InstanceAdminClient.Create();
+        InstanceName instanceName = InstanceName.FromProjectInstance(ProjectId, InstanceId);
+        try
+        {
+            Instance response = instanceAdminClient.GetInstance(instanceName);
+            return true;
+        }
+        catch (RpcException ex) when (ex.Status.StatusCode == StatusCode.NotFound)
+        {
+            CreateInstanceSample createInstanceSample = new CreateInstanceSample();
```

Comment:
Ping on this.

Reply:
Ping on this again, happy to defer to another PR, or if you think it's really a bad idea to have a test for create instance because of timeouts, etc., let me know. Else, I still think it's a good idea to have it, this line of code will almost never run in CI I think.

Reply:
Even if you just catch the "Instance already exists exception" the same way we have for CreateBackup, that's fine, it means the sample is working as intended.